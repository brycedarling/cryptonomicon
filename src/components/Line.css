.line {
  fill: none;
  shape-rendering: crispEdges;
  stroke: steelblue;
  stroke-width: 1;
}
