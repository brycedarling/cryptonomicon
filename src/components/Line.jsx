import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { line } from 'd3-shape';

class Line extends Component {
  static propTypes = {
    x: PropTypes.func.isRequired,

    y: PropTypes.func.isRequired,

    stroke: PropTypes.string,

    strokeWidth: PropTypes.number,

    data: PropTypes.arrayOf(PropTypes.shape({
      date: PropTypes.instanceOf(Date).isRequired,
      open: PropTypes.number.isRequired,
      high: PropTypes.number.isRequired,
      low: PropTypes.number.isRequired,
      close: PropTypes.number.isRequired,
      volume: PropTypes.number.isRequired,
      quoteVolume: PropTypes.number.isRequired,
      weightedAverage: PropTypes.number.isRequired,
    })).isRequired,
  };

  static defaultProps = {
    stroke: 'steelblue',
    strokeWidth: 1,
  };

  get line() {
    return line()
      .x(this.props.x)
      .y(this.props.y);
  }

  render() {
    return (
      <path
        className="line"
        style={{
          fill: 'none',
          shapeRendering: 'crispEdges',
          stroke: this.props.stroke,
          strokeWidth: this.props.strokeWidth,
        }}
        d={this.line(this.props.data)}
      />
    );
  }
}

export default Line;
