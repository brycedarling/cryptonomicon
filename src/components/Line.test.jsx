import React from 'react';
import renderer from 'react-test-renderer';
import Line from './Line';

describe('components', () => {
  describe('Line', () => {
    it('renders correctly', () => {
      const data = [{
        date: new Date('01/01/2017'),
        open: 1,
        close: 2,
        high: 3,
        low: 0,
        volume: 1,
        quoteVolume: 2,
        weightedAverage: 3,
      }, {
        date: new Date('02/01/2017'),
        open: 2,
        close: 3,
        high: 4,
        low: 1,
        volume: 2,
        quoteVolume: 3,
        weightedAverage: 4,
      }];

      const x = d => d.date;

      const y = d => d.volume;

      const tree = renderer.create((
        <Line
          x={x}
          y={y}
          data={data}
        />
      )).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
