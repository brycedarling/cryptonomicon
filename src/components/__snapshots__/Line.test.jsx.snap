// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Line renders correctly 1`] = `
<path
  className="line"
  d="M1483246800000,1L1485925200000,2"
  style={
    Object {
      "fill": "none",
      "shapeRendering": "crispEdges",
      "stroke": "steelblue",
      "strokeWidth": 1,
    }
  }
/>
`;
